cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/PhaseTapChangerConversion.java

```diff
@@ -335,41 +309,62 @@ private void addSteps(
         double xStepMin = xs[0];
         double xStepMax = xs[1];
 
-        double alphaMax = alphas.stream()
+        double angleMax = angles.stream()
                 .mapToDouble(Double::doubleValue)
                 .max()
                 .orElse(Double.NaN);
-        LOG.debug("ACTUAL    alphaMax {}", alphaMax);
+        LOG.debug("ACTUAL    angleMax {}", angleMax);
         LOG.debug("ACTUAL    xStepMin, xStepMax {}, {}", xStepMin, xStepMax);
 
-        // rho0 adjustment computed the same way that is done in TwoWindingsTransformer,
-        // using factor rho0square as a float
-        double rho0 = tx.getRatedU2() / tx.getRatedU1();
-        double rho0square = rho0 * rho0;
-        LOG.debug("ACTUAL    u2,u1,rho0square {}, {}, {}", tx.getRatedU2(), tx.getRatedU1(), rho0square);
+        // Take structural ratio
+        double ratio0 = tx.getRatedU2() / tx.getRatedU1();
```

Comment:
Why ````ratio0 = tx.getRatedU2() / tx.getRatedU1()```` ? I expected the inverse here.

Reply:
It is the structural ratio at end2 (after transmission impedance) so is ok.

Reply:
> It is the structural ratio at end2 (after transmission impedance) so is ok.

The structural ratio at end2 (after transmission impedance) is secondary nominal voltage / primary nominal voltage.
The structural ratio at end1 (before transmission impedance) is primary nominal voltage / secondary nominal voltage as you said but in the current CGMES conversion the structural ratio is always considered at end2 (after transmission impedance)

Reply:
Yes you are right, it is the ratio we use to "transport" the transmission impedance from side 1 to side 2.